### ISSUE TITLE
Ablaufdatum für Inventar Einbauen

### ISSUE THREAD

**Issue by GLSteiner** [Ferienhort/F.H.I.E.L.D.#16]
Auf Anfrage des Abenteuer Teams.

Inventar soll ein Datenfeld Ablaufjahr haben. Wenn dieses Jahr erreicht ist, dann markiert das System die abgelaufenen, ablaufenden (betreffenden) Stücke.

**Comment by oliveraustria** [comment]
Wir arbeiten aktuell mit der Idee, des LetztenNutzungsjahr statt dem Ablaufjahr, da wir glauben, dass dies in der Kommunikation noch klarer ist. Wäre es somit möglich, das Stück zu markieren, sobald das LetzteNutzungsjahr vorbei ist. Danke, das Abenteuerteam

**Comment by oliveraustria** [comment]
Aja, wir sind am evaluieren, ob neben dem Jahr auch das Quartal bzw. Monat relevant ist.

**Comment by GLSteiner** [comment]
Ok,

Also konfrom zu bleiben mit Rest, hier meine schneller Lösungsvorschlag:

- Neues Datenfeld unter erweitere Optionen (Grüner Pfeil Rechts)

- Sollte die Zeit nahe kommen, erscheint ein gelbes Rufzeichen links in der Listen Ansicht ist die Zeit überschritten werden ein rotes Rufzeichen. (Neben Stern, Uhr und Sprechblase, sehr unübersehbar)

- Ich würde Quartale oder Monate machen, da es wesentlich sinnvoller wird wenn euch das System Unterscheiden kann was bald abläuft geht und was schon ausgetauscht werden Sollte. Ein Warnhinweis der mit "Im laufe des Jahres... " beginnt, nehme ich auch nicht ernst.

**Comment by GLSteiner** [comment]
Das würde relativ schnell gehen und sich den anderen Funktionen eingliedern!

**Comment by GLSteiner** [comment]
Was haltet ihr von 2 Stichtagen: 

1. Juni und 1. September. 

Gelbes Rufzeichen (Warnzeichen...): Bis zum nächsten Stichtag austauschen
Rot: Stichtag ist schon vorbei!

**Comment by GLSteiner** [comment]
<img width="202" alt="screen shot 2018-01-30 at 7 32 05 am" src="https://user-images.githubusercontent.com/23318440/35551501-b4d28f92-058f-11e8-8107-c846adbb184e.png">
Ok, es gibt jetzt ein Feld in den erweiterten Funktionen für ein Ablaufdatum:

Herbst oder Frühjahr eines belieben Jahres.

- Es kann danach gefiltert werden
- Es wird ein gelbes Rufzeichen vor dem Namen wenn es bald abläuft
- Rotes Rufzeichen wenn es bereits abgelaufen ist
- Export nach Excel ist möglich
- Wird mitkopiert


<img width="748" alt="screen shot 2018-01-30 at 7 29 23 am" src="https://user-images.githubusercontent.com/23318440/35551429-547fcf56-058f-11e8-9ba7-31df0ca908a7.png">


Eingestellt wird es bei den erweiterten Optionen

<img width="302" alt="screen shot 2018-01-30 at 7 31 08 am" src="https://user-images.githubusercontent.com/23318440/35551474-9404569c-058f-11e8-836e-7c4d0afb893f.png">

**Comment by oliveraustria** [comment]
@GLSteiner Super danke, wobei ich ca. 5min gebraucht habe, um erweiterte Optionen zu finden. :) Also wenn du mal Zeit und Lust hast, würde ich dir für so manche Views gerne Anregungen geben.
Kann z.B. bootstrap sehr sehr empfehlen, seit kurzen ist auch Version 4 fertig.  Hier wäre ein Tabmenü, was ich für so einen Fall empfehlen würde: https://getbootstrap.com/docs/4.0/components/navs/#javascript-behavior

**Comment by oliveraustria** [comment]
Ups, du kannst gerne entscheiden, ob das ein reopening rechtfertigt. :)

**Comment by GLSteiner** [comment]
Bitte reopen alles was du willst 👍 

Bezuglich Bootstrap: kenn ich, wir verwenden es im FotoPortal. Problem ist nur, dass das FHIELD (zumindest die Weboberflache leider wirklich nicht dafür ausgerichtet ist. _Eigentlich_ wäre _eh_ ein kompletter UI Overhaul nötig - gefühlte 80% der Features die jetzt da sind, waren ursprünglich nicht vorgesehen und wurden _irgendwie_ reingehackt.

Bezüglich Views würde ich sehr gerne auf den Angebot zurück kommen! Die Funktionalität des Systems ist davon abhängig, wie gut und wie oft es verwendet wird, die oben erwähnten 80% gehen zumeist in die Richtung "Sinnvoll für User damit mehr Leute es verwenden" - Die Zeit beschränkt die Möglichkeiten hier aber eher auf rumschieben, umbennen und symbolwechsel, etc.

**Comment by oliveraustria** [comment]
Perfekt, mit rumschieben, umbenennen und symbolwechsel kann man ja schon viel erreichen. Aus meiner Sicht kann man ja auch Schritt für Schritt neue UI-Elemente einfügen. Ganz nachdem, wieviel Zeit und Lust du hast.

**Comment by GLSteiner** [comment]
Ja, also Verbesserungen wo mit wenig Aufwand viel erreicht werden kann werden eindeutig bevorzugt :)

Auch Schrittweise einführen finde ich gut. Problem ist beispielsweise, für diese Artikel Detailansicht müsste man aber eigentlich das gesamte iframe entfernen und alles auf eine zeitgemäße Methode neu machen. 

(ich bin wirklich nicht stolz drauf, da ist ziemlich viel "natürlich gewachsenes" dabei. Der Code der ursprünglichen Artikeleinstellungen ist auch wesentlich schöner als der der erweiterten Optionen....)

Auch wenn es vielleicht nicht offensichtlich ist ( 😄 ) ich bin wirklich kein UI Mensch: Ich freue mich sehr über Inputs, vor allem in dem Bereich. Go wild und eröffne da einfach Issues!

=> So für jetzt: Passt das Ablaufdatum so wie es ist? Kleinigkeit die mir hier mit den 2 Terminen aufgefallen ist: Wenn man im September nach den Dingen die "bald" ablaufen sucht, erscheinen die Dinge die im Frühling Ablaufen noch nicht als "bald", sondern nur die die im Herbst ablaufen.

Überlegungen:

- Herbst/Frühling so anpassen, dass ihre Bedeutung ähnlicher wird.
- Herbst/Frühling durch eine "inkl. Sommer 20XX" oder "exkl. Sommer 20XX" Option ersetzen. 
- Oder einfach das Feld so umbeschriften: "Hält bis inklusive Aktion XXXX" und somit nur die Eingabe einer Jahreszahl ermöglichen und inklusive Zeitangaben zum FHStandart erklären.

**Comment by oliveraustria** [comment]
Also ich würde auch kein totales Redesign machen, aus meiner Erfahrung kann man bootstrap auch sehr gut schritt für schritt einbauen.

Ich würde es am besten finden, wenn man einfach ein Monat angeben muss und dann kann man sich als Team überlegen, wie man es macht.

**Comment by GLSteiner** [comment]
Bist du dir sicher, das Monat gut ist?

Es macht nämlich das Filtern und die Warnzeichen sowohl komplizierter als auch intransparenter: Ich weiß nicht ob sich das auszahlt, wenn die Nutzer vermutlich am häufigsten nur fragen: "was müssen wir bis nächstes Jahr ersetzen?"

Ich glaub schon fast, dass unmissverständliche MVP ist "Hält bis inklusive Sommer XXXX".
